src/DDTrace/Integrations/Guzzle/GuzzleIntegration.php

```diff
@@ -3,30 +3,159 @@
 
 namespace DDTrace\Integrations\Guzzle;
 
+use DDTrace\Configuration;
+use DDTrace\Contracts\Span;
+use DDTrace\Format;
+use DDTrace\GlobalTracer;
+use DDTrace\Http\Urls;
 use DDTrace\Integrations\Integration;
+use DDTrace\Tag;
+use DDTrace\Type;
 use DDTrace\Util\CodeTracer;
-use DDTrace\Util\Versions;
-use DDTrace\Integrations\Guzzle\V5\GuzzleIntegrationLoader as V5Loader;
-use DDTrace\Integrations\Guzzle\V6\GuzzleIntegrationLoader as V6Loader;
 
 final class GuzzleIntegration
 {
     const NAME = 'guzzle';
 
+    /**
+     * @var CodeTracer
+     */
+    private $codeTracer;
+
+    public function __construct()
+    {
+        $this->codeTracer = CodeTracer::getInstance();
+    }
+
     public static function load()
     {
-        if (!defined('GuzzleHttp\ClientInterface::VERSION')) {
-            return Integration::NOT_LOADED;
+        $instance = new self();
+        $instance->doLoad();
+        return Integration::LOADED;
+    }
+
+    /**
+     * @return int
+     */
+    public function doLoad()
+    {
+        $self = $this;
+
+        $postCallback = function (Span $span, $response) use ($self) {
+            $self->setStatusCodeTag($span, $response);
+        };
+
+        $this->codeTracer->tracePublicMethod(
+            'GuzzleHttp\Client',
+            'send',
+            $this->buildPreCallback('send'),
+            $postCallback
+        );
+        $this->codeTracer->tracePublicMethod(
+            'GuzzleHttp\Client',
+            'transfer',
+            $this->buildPreCallback('transfer'),
+            $postCallback
+        );
+
+        return Integration::LOADED;
+    }
+
+    /**
+     * @param string $method
+     * @return \Closure
+     */
+    private function buildPreCallback($method)
+    {
+        $self = $this;
+        return function (Span $span, array $args) use ($self, $method) {
+            list($request) = $args;
+            $self->applyDistributedTracingHeaders($span, $request);
+            $span->setTag(Tag::SPAN_TYPE, Type::HTTP_CLIENT);
+            $span->setTag(Tag::SERVICE_NAME, GuzzleIntegration::NAME);
+            $span->setTag(Tag::HTTP_METHOD, $request->getMethod());
+            $self->setUrlTag($span, $request);
+            $span->setTag(Tag::RESOURCE_NAME, $method);
+        };
+    }
+
+    /**
+     * @param Span $span
+     * @param mixed $request
+     */
+    protected function setUrlTag(Span $span, $request)
```

Comment:
Just a nit-pick, but these can be marked `private` since it's a `final` class. :)